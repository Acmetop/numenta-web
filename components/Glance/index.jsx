// Numenta shared web components
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import {trims} from '../../utils/shared'

import DefineItem from '../DefineItem'
import DefineList from '../DefineList'
import DefineTitle from '../DefineTitle'
import Logo from '../Logo'
import Spacer from '../Spacer'
import Tag from '../Tag'
import TextLink from '../TextLink'

import styles from './index.css'


/**
 * At-A-Glance React view component. Quick Details about the company/site.
 */
const Glance = (props, context) => {
  const {config} = context
  const {links} = config
  const data = {
    About: trims`Developing a neocortical theory for biological and machine
            intelligence.`,
    Established: (
      <span>
        Feb 4, 2005
        <Spacer />
        Redwood City, CA
      </span>
    ),
    Employees: '15 — 20',
    Business: (
      <TextLink to="/assets/pdf/apps/licensing-guide.pdf">Licensing</TextLink>
    ),
    Content: (
      <span>
        <TextLink to={links.in.blog}>Blog</TextLink>
        <Spacer />
        <TextLink to={links.in.newsletter}>Newsletter</TextLink>
      </span>
    ),
    Press: (
      <span>
        <TextLink to={links.in.press}>
          Releases
        </TextLink>
        <Spacer />
        <TextLink to={`${links.in.press}#links`}>
          Links
        </TextLink>
      </span>
    ),
    Events: (
      <TextLink to={links.in.events}>Events</TextLink>
    ),
    /* eslint-disable max-len */
    Latest: (
      <TextLink to="/blog/2016/09/02/why-did-we-completely-change-our-website-design-the-story-behind-our-new-look/">
        Why did we overhaul our web design? The story behind our new look. {' '}
        <Tag>New</Tag>
      </TextLink>
    ),
    /* eslint-ensable max-len */
  }
  const defs = []

  for (const key in data) {
    if ({}.hasOwnProperty.call(data, key)) {
      defs.push(
        <DefineTitle key={`DefineTitle-${key}`}>
          {key}
        </DefineTitle>
      )
      defs.push(
        <DefineItem key={`DefineItem-${key}`}>
          {data[key]}
        </DefineItem>
      )
    }
  }

  return (
    <aside className={styles.glance}>
      <div className={styles.logo}>
        <Logo />
      </div>
      <DefineList direction="horizontal">
        {defs}
      </DefineList>
    </aside>
  )
}

Glance.contextTypes = {
  config: React.PropTypes.object,
}

export default Glance
