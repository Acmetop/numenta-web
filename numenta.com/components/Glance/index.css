._rules dl > dt,
._rules dl > dd {
  border-top: 1px solid #ddd;
}


.glance {
  composes: bg-near-white from 'tachyons';
  composes: br2 from 'tachyons';
  composes: f5 from 'tachyons';
  composes: f6-ns from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: pb3 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: pt2 from 'tachyons';
  composes: tl from 'tachyons';
  composes: _rules;
}

.logo {
  composes: center from 'tachyons';
  composes: mv2 from 'tachyons';
  composes: w5 from 'tachyons';
  composes: w4-m from 'tachyons';
}
