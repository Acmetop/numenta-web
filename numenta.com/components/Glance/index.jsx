import React from 'react'

import {trims} from '../../utils/shared'

import DefineItem from '../DefineItem'
import DefineList from '../DefineList'
import DefineTitle from '../DefineTitle'
import Logo from '../Logo'
import Spacer from '../Spacer'
import Tag from '../Tag'
import TextLink from '../TextLink'

import styles from './index.css'


const Glance = (props, context) => {
  const {config} = context
  const {links} = config
  const data = {
    About: trims`Numenta develops machine intelligence technology based on
            neocortical theory.`,
    Established: 'Feb 4, 2005',
    Location: 'Redwood City, CA',
    Employees: '15 â€” 20',
    Business: 'Licensing',
    Type: 'Private',
    Content: (
      <span>
        <TextLink to={links.in.blog}>Blog</TextLink>
        <Spacer />
        <TextLink to={links.in.newsletter}>Newsletter</TextLink>
      </span>
    ),
    Press: (
      <span>
        <TextLink to={links.in.press}>
          Releases
        </TextLink>
        <Spacer />
        <TextLink to={`${links.in.press}#links`}>
          Links
        </TextLink>
      </span>
    ),
    Events: (
      <TextLink to={links.in.events}>Events</TextLink>
    ),
    Latest: (
      <TextLink to={links.in.htmstudio}>
        Numenta releases HTM Studio! <Tag>New</Tag>
      </TextLink>
    ),
  }
  let defs = []

  for (let key in data) {
    if ({}.hasOwnProperty.call(data, key)) {
      defs.push(
        <DefineTitle key={`DefineTitle-${key}`}>
          {key}
        </DefineTitle>
      )
      defs.push(
        <DefineItem key={`DefineItem-${key}`}>
          {data[key]}
        </DefineItem>
      )
    }
  }

  return (
    <aside className={styles.glance}>
      <div className={styles.logo}>
        <Logo />
      </div>
      <DefineList direction="horizontal">
        {defs}
      </DefineList>
    </aside>
  )
}

Glance.contextTypes = {
  config: React.PropTypes.object,
}

export default Glance
