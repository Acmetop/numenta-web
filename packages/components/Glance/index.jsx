// Numenta shared web components
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'
import {trims} from 'numenta-web-shared-utils/shared'

import DefineItem from '../DefineItem'
import DefineList from '../DefineList'
import DefineTitle from '../DefineTitle'
import Spacer from '../Spacer'
import Tag from '../Tag'
import TextLink from '../TextLink'

import styles from './index.css'


/**
 * At-A-Glance React view component. Quick Details about the company/site.
 */
const Glance = ({children, head}, context) => {
  const {config} = context
  const {links} = config

  return (
    <aside className={styles.glance}>
      <div>{head}</div>
      <DefineList direction="horizontal">
        {children}
      </DefineList>
    </aside>
  )
}

Glance.contextTypes = {
  config: React.PropTypes.object,
}

Glance.propTypes = {
  children: React.PropTypes.array.isRequired,
  head: React.PropTypes.element.isRequired,
}

export default Glance
