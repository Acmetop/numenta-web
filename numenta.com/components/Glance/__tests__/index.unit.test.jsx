import React from 'react'
import renderer from 'react-test-renderer'

import Glance from '../../Glance'


describe('<Glance /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Glance />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
