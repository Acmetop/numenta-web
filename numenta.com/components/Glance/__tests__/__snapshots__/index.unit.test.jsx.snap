exports[`Glance React component Renders correctly 1`] = `
<aside
  className="glance">
  <div
    className="logo">
    <img
      alt="Numenta Logo"
      className="image"
      onClick={undefined}
      src="filename.ext"
      title="Numenta" />
  </div>
  <dl
    className="defineList horizontal">
    <dt
      className="defineTitle horizontal">
      About
    </dt>
    <dd
      className="defineItem horizontal">
      Developing a neocortical theory for biological and machine intelligence.
    </dd>
    <dt
      className="defineTitle horizontal">
      Established
    </dt>
    <dd
      className="defineItem horizontal">
      <span>
        Feb 4, 2005
        <span
          className="spacer">
          •
        </span>
        Redwood City, CA
      </span>
    </dd>
    <dt
      className="defineTitle horizontal">
      Employees
    </dt>
    <dd
      className="defineItem horizontal">
      15 — 20
    </dd>
    <dt
      className="defineTitle horizontal">
      Business
    </dt>
    <dd
      className="defineItem horizontal">
      <a
        className="textlink"
        href="/assets/pdf/apps/licensing-guide.pdf"
        onClick={[Function instrumentOnClick]}
        target={undefined}>
        Licensing
      </a>
    </dd>
    <dt
      className="defineTitle horizontal">
      Content
    </dt>
    <dd
      className="defineItem horizontal">
      <span>
        <a
          className="textlink"
          href={undefined}
          onClick={[Function bound handleClick]}
          style={Object {}}
          target={undefined}>
          Blog
        </a>
        <span
          className="spacer">
          •
        </span>
        <a
          className="textlink"
          href={undefined}
          onClick={[Function bound handleClick]}
          style={Object {}}
          target={undefined}>
          Newsletter
        </a>
      </span>
    </dd>
    <dt
      className="defineTitle horizontal">
      Press
    </dt>
    <dd
      className="defineItem horizontal">
      <span>
        <a
          className="textlink"
          href={undefined}
          onClick={[Function bound handleClick]}
          style={Object {}}
          target={undefined}>
          Releases
        </a>
        <span
          className="spacer">
          •
        </span>
        <a
          className="textlink"
          href={undefined}
          onClick={[Function bound handleClick]}
          style={Object {}}
          target={undefined}>
          Links
        </a>
      </span>
    </dd>
    <dt
      className="defineTitle horizontal">
      Events
    </dt>
    <dd
      className="defineItem horizontal">
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Events
      </a>
    </dd>
    <dt
      className="defineTitle horizontal">
      Latest
    </dt>
    <dd
      className="defineItem horizontal">
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Why did we overhaul our web design? The story behind our new look. 
         
        <span
          className="tag">
          New
        </span>
      </a>
    </dd>
  </dl>
</aside>
`;
